src/components/dialog/content/SubscriptionRequiredDialogContent.vue

```diff
@@ -0,0 +1,125 @@
+<template>
+  <div class="subscription-required-dialog flex flex-col gap-4 p-6">
+    <div class="flex flex-col items-center gap-4 text-center">
+      <i class="pi pi-lock text-6xl text-primary-400" />
+      <h2 class="text-2xl font-bold">
+        {{ $t('subscription.required.title') }}
+      </h2>
+      <p class="text-muted">
+        {{
+          isPolling
+            ? $t('subscription.required.waitingForSubscription')
+            : $t('subscription.required.message')
+        }}
+      </p>
+    </div>
+
+    <div
+      class="flex flex-col gap-3 rounded-lg border p-4 dark-theme:bg-charcoal-600"
+    >
+      <SubscriptionBenefits />
+    </div>
+
+    <Button
+      :label="$t('subscription.required.subscribe')"
+      class="w-full"
+      :loading="isLoading"
+      :disabled="isPolling"
+      @click="handleSubscribe"
+    />
+
+    <Button
+      :label="$t('g.cancel')"
+      text
+      severity="secondary"
+      class="self-center"
+      :disabled="isPolling"
+      @click="handleCancel"
+    />
+  </div>
+</template>
+
+<script setup lang="ts">
+import Button from 'primevue/button'
+import { onBeforeUnmount, ref } from 'vue'
+
+import SubscriptionBenefits from '@/components/common/SubscriptionBenefits.vue'
+import { useFirebaseAuthActions } from '@/composables/auth/useFirebaseAuthActions'
+
+const emit = defineEmits<{
+  close: [subscribed: boolean]
+}>()
+
+const authActions = useFirebaseAuthActions()
+const isLoading = ref(false)
+const isPolling = ref(false)
+let pollInterval: number | null = null
+
+const POLL_INTERVAL_MS = 3000 // Poll every 3 seconds
+const MAX_POLL_DURATION_MS = 5 * 60 * 1000 // Stop polling after 5 minutes
+
+const startPollingSubscriptionStatus = () => {
+  isPolling.value = true
+  isLoading.value = true
+
+  const startTime = Date.now()
+
+  const poll = async () => {
+    try {
+      if (Date.now() - startTime > MAX_POLL_DURATION_MS) {
+        stopPolling()
+        return
+      }
+
+      const status = await authActions.fetchSubscriptionStatus()
+
+      if (status?.is_active) {
+        stopPolling()
+        emit('close', true)
+      }
+    } catch (error) {
+      console.error(
+        '[SubscriptionRequiredDialog] Error polling subscription status:',
+        error
+      )
+    }
+  }
+
+  void poll()
+  pollInterval = window.setInterval(poll, POLL_INTERVAL_MS)
+}
+
+const stopPolling = () => {
+  if (pollInterval) {
+    clearInterval(pollInterval)
+    pollInterval = null
+  }
+  isPolling.value = false
+  isLoading.value = false
+}
```

Comment:
I think [useTimeoutPoll](https://vueuse.org/core/useTimeoutPoll/#usetimeoutpoll) could handle most of this.